app/Http/Controllers/ProductController.php

```diff
@@ -96,15 +97,28 @@ public function destroy(Product $product): \Illuminate\Http\JsonResponse
      * Get all products with specified location.
      *
      * @param Location $location
-     * @return \Illuminate\Http\Resources\Json\AnonymousResourceCollection
+     * @return AnonymousResourceCollection
      */
-    public function getProductsByLocation(Location $location): \Illuminate\Http\Resources\Json\AnonymousResourceCollection
+    public function getProductsByLocation(Location $location): AnonymousResourceCollection
     {
 
-        $products = Product::with(['location_products' => function($query) use ($location){
+        $products = Product::with(['location_products' => function ($query) use ($location) {
             return $query->where('location_id', $location->id);
         }])->get();
 
         return ProductResource::collection($products);
     }
+
+    /**
+     * Search product by name or barcode.
+     *
+     * @param string $search
+     * @return AnonymousResourceCollection
+     */
+    public function search(string $search): AnonymousResourceCollection
+    {
+        $product = Product::search($search)->get();
+
+        return ProductResource::collection($product);
+    }
```

Comment:
Search should probably have a quick search function for search bars to return just the top 10 results. The "deeper" search should probably make use of the paginate function or similar.

Reply:
I have added pagiantion to the search, so if it is only needed to get small amount we can